libs/ledger-live-common/src/wallet-api/Exchange/server.ts

```diff
@@ -97,78 +104,34 @@ export const handlers = ({
         tracking.startExchangeNoParams(manifest);
         return { transactionId: "" };
       }
-      if ("fromAccountId" in params) {
-        const realFromAccountId = getAccountIdFromWalletAccountId(params.fromAccountId);
-        if (!realFromAccountId) {
-          return Promise.reject(new Error(`accountId ${params.fromAccountId} unknown`));
-        }
 
-        const fromAccount = accounts.find(acc => acc.id === realFromAccountId);
+      let exchangeParams: ExchangeStartParamsUiRequest = {
+        exchangeType: params.exchangeType,
+      };
 
-        if (!fromAccount) {
-          throw new ServerError(createAccountNotFound(params.fromAccountId));
+      switch (params.exchangeType) {
+        case "SWAP": {
+          exchangeParams = extractSwapStartParam(params as ExchangeStartSwapParams, accounts);
```

Comment:
I think we could remove the `as` keyword here by using a guard function in the case check that takes `params` as a parameter to use the return type `_params is ExchangeStartSwapParams` (the underscore is needed if we don't use params in the guard function)

Now that I think about it, another and maybe simpler solution might just be to use a `if` instead of the `switch` so typescript can type narrow correctly

Reply:
That's a shame Typescript is not able to type narrow on `switch` :(